### PR TITLE
Minor bug fixes & optimizations

### PR DESCRIPTION
**Pull request by TheAnkurGoswami**
## What does this PR do?
Fixes minor bugs

## Changes

* Added a functionality to remove manually downloaded data when uninstalling/updating the library.
* Fixed the red(partial termination) progress bar issue (Reason: When calling the download function again, the `expected_records` was not getting reset. So it was taking previous number of records into account. That caused total number of signals to have a wrong value. That's why the data was downloading properly but the wrong total raised red progress bar error.)

## Before reviewing

- [x] Did you test this PR?
- [ ] Is there any migration or seeding necessary?
- [ ] If a new field is added to signals in the api, did you update the sample test files?

### Links:

- Jira: https://myaccern.atlassian.net/browse/RSL-212

### CODE REVIEW

File: packages/python/accern_data/util.py [hunk 1]
```diff
@@ -119,6 +121,25 @@ def generate_file_response(
     return response_obj
 
 
+def register_on_record() -> None:
+    directory, folder = os.path.split(os.path.split(__file__)[0])
+    full_directory = os.path.join(
```
Comment (JosuaKrause) [inline packages/python/accern_data/util.py:126]: is this path guaranteed? use __file__

File: packages/python/accern_data/__init__.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
-__version__ = "0.0.2-rc1"
+__version__ = "0.0.2rc1"
```
[the remark this answers is not in the document]
Comment (TheAnkurGoswami) [reply packages/python/accern_data/__init__.py:1]: the hyphen -- checked that with accern-xyme and thought that hyphen is not a good thing to put

File: packages/python/accern_data/util.py (hunk 1 above)
Comment (TheAnkurGoswami) [reply packages/python/accern_data/util.py:126]: file?

File: packages/python/accern_data/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JosuaKrause) [reply packages/python/accern_data/__init__.py:1]: see other pr

File: packages/python/accern_data/util.py (hunk 1 above)
Comment (JosuaKrause) [reply packages/python/accern_data/util.py:126]: `__file__`

Comment (TheAnkurGoswami) [reply packages/python/accern_data/util.py:126]: i'm using that. no?

File: packages/python/accern_data/accern_data.py [hunk 3]
```diff
@@ -365,6 +365,9 @@ def __init__(
         self._first_error = True
         self._expected_records: List[int] = []
 
+    def reset(self) -> None:
+        self._expected_records = []
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:369]: why do we store this in the object in the first place? it's only local info for the current download so it should just be a variable in the function

Comment (TheAnkurGoswami) [reply packages/python/accern_data/accern_data.py]: right!

File: packages/python/accern_data/util.py [hunk 4]
```diff
@@ -147,6 +149,25 @@ def generate_file_response(
...
+        directory,
+        f"{folder}-{__version__.replace('-', '')}.dist-info",
+        "RECORD")
+    lines = {
+        f"{folder}/data/data-2022.json",
+        f"{folder}/data/data-2022.csv",
+        f"{folder}/data/.DS_Store",
```
Comment (JosuaKrause) [inline packages/python/accern_data/util.py:161]: those files will only exist on mac -- you have to check if they exist before adding them to the list

File: packages/python/accern_data/accern_data.py [hunk 5]
```diff
@@ -572,6 +580,8 @@ def download_range(
         if output_path is None:
             output_path = "./"
         os.makedirs(output_path, exist_ok=True)
+        expected_records: List[int] = []
```
[the remark this answers is not in the document]
Comment (TheAnkurGoswami) [reply packages/python/accern_data/accern_data.py:576]: this is the one we'll need

File: packages/python/accern_data/accern_data.py [hunk 6]
```diff
@@ -633,7 +636,8 @@ def download_range(
                     filters=valid_filters,
                     progress_bar=progress_bar)
         progress_bar.close()
-        self._expected_records = []
+        if len(self._error_list) > 0:
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:639]: Only if it was an actual error. If we suppress errors we don't print them here. The user has to check

File: README.md [hunk 7]
```diff
@@ -68,3 +68,13 @@ client.download_range(
 ```python
 accern_data.create_client("http://api.example.com/", "SomeRandomToken").download(start_date="2022-01-03", output_path="./", output_pattern="data", end_date="2022-03-04", mode="csv", filters={"entity_ticker": "HURC"})
 ```
+
+
+### Error logging:
+
+While downloading the data, any critical error will get raised.
```
Comment (JosuaKrause) [inline README.md:75]:
```suggestion
While downloading the data any critical error will get raised.
```

**Review by JosuaKrause — APPROVED** [review]
lgtm

**Pull request merged**